feat: 헤더 UI 및 오른쪽 사이드바 UI 추가, 라우터 설정

jihyeon-songb opened the pull request:
## ✨ 개요

<!-- 이번 PR에서 무엇을 했는지 간단히 설명 (ex: 회원가입 이메일 중복 검사 기능 추가) -->
헤더 UI와 오른쪽 사이드바 UI를 추가하고, 라우터 및 폰트/아이콘 관련 설정을 적용

## 💡 변경 내용

<!-- 주요 변경사항을 요약해서 bullet point로 작성해주세요. -->

- 헤더 컴포넌트 UI 추가 및 전체 CSS 초기화
- react-router 라이브러리 설치 및 기본 세팅
- 불필요한 .tsx 코드 제거 및 로고/프로필 파일 이동
- vite-plugin-svgr 설치 및 SVG 아이콘 추가
- Pretendard 글꼴 적용 및 TailwindCSS 커스텀 색상 추가
- 오른쪽 사이드바 컴포넌트 UI 추가
- 메인레이아웃 별도 분리
- Prettier 설치 및 설정 파일 추가
- 함수 선언, self-closing 코드 스타일 통일


## 📸 스크린샷 (선택 사항)

<!-- UI/UX 변경 사항이 있을 경우, 관련 스크린샷이나 GIF를 Before / After 캡처를 첨부해주세요. -->
<img width="1509" height="819" alt="image" src="https://github.com/user-attachments/assets/14ad2708-e305-49a6-b474-0be831e9c786" />


## 🧩 관련 이슈

<!-- PR과 관련된 이슈 번호를 링크하세요. -->

- Issue: 
Closes #14
Closes #15 
Closes #25  

## 🧐 리뷰어에게

<!-- 리뷰 시 중점적으로 확인할 부분이나 주의할 점을 작성해주세요 -->
컴포넌트 구조와 라우터 세팅 방식, 레이아웃 분리 방식이 적절한지 확인 부탁드립니다.

## ✅ Checklist

- [x]  커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] Self-Review를 완료했습니다.
- [x] 문서가 필요한 경우, 관련 문서를 업데이트했습니다.

jihyeon-songb committed:
feat: 헤더 컴포넌트 추가 및 전체 css 초기화

- 헤더 컴포넌트 UI 추가
- 로고, 프로필 이미지 추가
- App.css 내용 삭제
- 전체 css 초기화

Related to: #14

jihyeon-songb committed:
feat: react-router 라이브러리 설치 및 세팅

Related to: #14

jihyeon-songb committed:
chore: .tsx 작성한 코드 삭제 및 로고, 프로필 파일 이동

- .tsx 작성한 코드 삭제
- 로고, 프로필 파일 public 폴더에서 src/assets 폴더로 이동

Related to: #14

jihyeon-songb committed:
feat: vite-plugin-svgr 설치 및 svg 아이콘 추가

- vite-plugin-svgr 설치 및 세팅
- 사이드바 컴포넌트 svg 아이콘 추가

jihyeon-songb committed:
feat: Pretendard 글꼴 변경 및 tailwindCSS 커스텀 색상 추가

jihyeon-songb committed:
feat: 오른쪽 사이드바 컴포넌트 UI 추가

Related to: #15

jihyeon-songb committed:
refactor: App 내부 레이아웃을 별도 파일로 분리

jihyeon-songb committed:
refactor: main 페이지 코드 정리

pro-flab commented on src/layouts/MainLayout.tsx:
```diff
@@ -0,0 +1,21 @@
+    return (
+        <div className="flex">
+            <div className="flex flex-col w-full min-w-0 flex-1">
+                <Header></Header>
```
```suggestion
                <Header />
```

pro-flab replied:
```tsx
 <Header></Header>
```
children 영역이 비어 있을 경우에는 <Header />처럼 쓰는게 좋습니다.

pro-flab commented on src/layouts/MainLayout.tsx:
```diff
@@ -0,0 +1,21 @@
+    children: ReactNode;
+};
+
+const MainLayout: FC<LayoutProps> = ({ children }) => {
```
React에서 컴포넌트를 정의할 수 있는 방법이 여러 가지가 있어요.

1.
```tsx
const Component: FC<Props> = () => {
   return <></>
}
```


2.
```tsx
const Component = (props: Props) => {
   return <></>
}
```

3.

```tsx
function Component(props: Props) {
   return <></>
}
```

pro-flab commented on src/components/common/SideBar.tsx:
```diff
@@ -0,0 +1,65 @@
+    );
+}
+
+export default function SideBar() {
```
전체적인 코드 일관성을 위해 컴포넌트 선언 방식을 통일하면 좋겠습니다.

pro-flab commented on src/components/common/SideBar.tsx:
```diff
@@ -0,0 +1,65 @@
+    return (
+        <div
+            data-open={isOpenSidebar}
+            className={`group/sidebar relative z-10 flex h-dvh overflow-hidden
+                  ${isOpenSidebar ? "w-[370px]" : "w-[56px]"}
+                  transition-[width] duration-200 ease-in-out`}
```
shadcn 설치 시 설치되는 cn이라는 유틸리티를 잘 활용해보면 좋을 것 같아요!

```tsx
className={cn('relative z-10', isOpenSidebar ? 'w-[370px]' : 'w-[56px]', isOk ? 'h-4' : null)}
```

pro-flab commented on src/components/common/Header.tsx:
```diff
@@ -0,0 +1,29 @@
+import {Button} from "@/components/ui/button";
+import {useState} from "react";
+
+export default function Header() {
```
함수 선언 형식 통일 부탁드립니다!

pro-flab commented on src/components/common/Header.tsx:
```diff
@@ -0,0 +1,29 @@
+                    </ul>
+                </div>
+                <div>
+                    { isLogin ?
+                        <Button variant="default" size="sm">로그인</Button> :
+                        <img className="rounded-md" src="/src/assets/profile.webp" width="32" height="32" alt="프로필이미지"/>
+                    }
```
VSCode에서 Cmd+S(저장) 시 포맷팅이 되도록 세팅을 해줘야할 것 같습니다. 이 부분은 코드 포맷 통일에 중요한 부분이니 신경 써주시기 바랍니다!

jihyeon-songb committed:
style: 코드 스타일 통일 (함수 선언, self-closing, cn)

jihyeon-songb committed:
chore: Prettier 설정 파일 추가

- 2스페이스 들여쓰기, 작은따옴표, 세미콜론
- trailing comma all, arrow parens always
- prettier-plugin-tailwindcss 적용
